server/repository/src/db_diesel/report_query.rs

```diff
@@ -0,0 +1,276 @@
+use crate::{RepositoryError, StorageConnection};
+#[cfg(feature = "postgres")]
+use diesel::sql_types::*;
+
+#[cfg(feature = "postgres")]
+#[derive(QueryableByName, Debug, PartialEq)]
+pub struct JsonDataRow {
+    #[sql_type = "Text"]
+    data: String,
+}
+
+#[cfg(feature = "postgres")]
+pub fn query_json(
+    connection: &StorageConnection,
+    sql: &str,
+    parameters: &serde_json::Map<String, serde_json::Value>,
+) -> Result<Vec<serde_json::Value>, RepositoryError> {
+    use diesel::connection::SimpleConnection;
+    use diesel::{sql_query, RunQueryDsl};
+    use regex::Regex;
+    use util::uuid::small_uuid;
+
+    // extract all used params from the sql query string, e.g. $myVariable
+    let re = Regex::new(r"(\$[a-zA-Z0-9]+)").unwrap();
+    // stores the variable name and the found parameter value, e.g. ($myVariable, "Hello")
+    let mut used_params = Vec::<(String, serde_json::Value)>::new();
+    for param in re.captures_iter(sql) {
+        let param: &str = &param[0];
+        if used_params.iter().any(|it| &it.0 == param) {
+            continue;
+        }
+        let param_name = &param[1..];
+        let Some(param_value) = parameters.get(param_name) else {
+            return Err(RepositoryError::DBError {
+                msg: format!("Invalid parameter: {param_name}"),
+                extra: "".to_string(),
+            });
+        };
+        used_params.push((param.to_string(), param_value.clone()))
+    }
+
+    // remove trailing ";" if there is any
+    let mut sql = if sql.chars().last() == Some(';') {
+        sql[..sql.len() - 1].to_string()
+    } else {
+        sql.to_string()
+    };
+    // Replace named variable like $myVariable with the numbered parameters like $1. Using the order
+    // in which variables where first used.
+    for (i, param) in used_params.iter().enumerate() {
+        sql = sql.replace(&param.0, &format!("${}", i + 1));
+    }
+    // Create the string containing all the parameter values
+    let param_values = used_params
+        .iter()
+        .map(|it| match &it.1 {
+            serde_json::Value::Null => "NULL".to_string(),
```

Comment:
Can you give some more info please? don't see to problem...

Reply:
I was imagining a query like 

`SELECT * FROM invoice WHERE some_filed = $someField`

If you passed the parameters
`{"someField":" null}`

What would be the expected behaviour?

Reply:
I would expect `SELECT * FROM invoice WHERE some_filed = NULL` (that was my intention)

Reply:
My point is `SELECT * FROM invoice WHERE  comment is NULL` would return the list of invoices where the comments are null, but `SELECT * FROM invoice WHERE  comment = NULL` will not return anything.

It's not exactly wrong, I'm just imagining it could easily catch us out.

Reply:
ok thanks, but this shouldn't affect the code here, e.g. the user still has to write either `= $value` or `is $value`. This code just stupidly replaces the named parameter with whatever was provided.